src/OrbitGl/TimelineUi.h

```diff
@@ -32,6 +32,16 @@ class TimelineUi : public CaptureViewElement {
                     uint64_t max_timestamp_ns) const;
   void RenderMargin(Batcher& batcher) const;
   void RenderBackground(Batcher& batcher) const;
+  [[nodiscard]] std::string GetLabel(uint64_t tick_ns,
+                                     uint32_t number_of_decimal_places_needed) const;
+  [[nodiscard]] std::vector<uint64_t> GetTicksForNonOverlappingLabels(
+      TextRenderer& text_renderer, const std::vector<uint64_t>& all_major_ticks,
```

Comment:
Will `const TextRenderer&` also work as a parameter here? I get why this is needed, it just feels a bit weird to have a mutable TextRenderer as a method that only calculates without doing any rendering.

Reply:
We have a strange method MaybeLoadAndGetGlyph, not const (because it could be loading a glyph) that it used to Get the Width of a string. It doesn't have any sense in my opinion, but I didn't make it and I don't fully understand the consequences.

An easy fix might be making texture_atlas_changed_ mutable, but anyway as you suggested, not part of this PR.

Reply:
Ah, yeah, that's what I suspected... thanks for checking.